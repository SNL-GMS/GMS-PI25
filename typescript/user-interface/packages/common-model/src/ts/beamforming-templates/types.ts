import type { CommonTypes, EventTypes, SignalDetectionTypes } from '../common-model';
import type { VersionReference } from '../faceted';
import type { FilterDefinition } from '../filter';
import type {
  Channel,
  Orientation
} from '../station-definitions/channel-definitions/channel-definitions';
import type { Station } from '../station-definitions/station-definitions/station-definitions';

export enum SamplingType {
  SNAPPED = 'SNAPPED',
  INTERPOLATED = 'INTERPOLATED'
}

export enum BeamSummation {
  COHERENT = 'COHERENT',
  INCOHERENT = 'INCOHERENT',
  RMS = 'RMS'
}

export enum BeamType {
  CONTINUOUS_LOCATION = 'CONTINUOUS_LOCATION',
  DETECTION = 'DETECTION',
  EVENT = 'EVENT',
  FK = 'FK'
}

export enum InterpolationMethod {
  NEAREST_SAMPLE = 'Nearest Sample',
  INTERPOLATED = 'Interpolated'
}

export interface BeamDescription {
  readonly beamSummation: BeamSummation;
  readonly beamType: BeamType;
  readonly phase: string;
  readonly samplingType: SamplingType;
  readonly twoDimensional: boolean;
  readonly prefilterDefinition?: FilterDefinition;
}

export interface BeamformingTemplate {
  readonly beamDescription: BeamDescription;
  readonly inputChannels: VersionReference<'name'>[] | Channel[];
  readonly minWaveformsToBeam: number;
  readonly orientationAngleToleranceDeg: number;
  readonly sampleRateToleranceHz: number;
  readonly station: VersionReference<'name'> | Station;
  readonly beamDuration?: number;
  readonly leadDuration?: number;
}

export interface BeamParameters {
  eventHypothesis: EventTypes.EventHypothesis;
  location: CommonTypes.Location;
  minWaveformsToBeam: number;
  orientationAngles: Orientation;
  orientationAngleToleranceDeg: number;
  receiverToSourceAzimuthDeg: number;
  sampleRateHz: number;
  sampleRateToleranceHz: number;
  signalDetectionHypothesis: SignalDetectionTypes.SignalDetectionHypothesis;
  slownessSecPerDeg: number;
}

export interface BeamDefinition {
  beamDescription: BeamDescription;
  beamParameters: BeamParameters;
}

export type BeamformingTemplatesByPhase = Record<string, BeamformingTemplate>;

export type BeamformingTemplatesByStationByPhase = Record<string, BeamformingTemplatesByPhase>;

export type BeamformingTemplatesByBeamTypeByStationByPhase = Record<
  string,
  BeamformingTemplatesByStationByPhase
>;
