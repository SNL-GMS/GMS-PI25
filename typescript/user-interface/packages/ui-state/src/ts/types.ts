import type {
  ChannelSegmentTypes,
  ChannelTypes,
  CommonTypes,
  EventTypes,
  FilterTypes,
  FkTypes,
  SignalDetectionTypes
} from '@gms/common-model';
import type { AssociationConflict, Event } from '@gms/common-model/lib/event';
import type {
  FilterDefinition,
  FilterDefinitionByFilterDefinitionUsage
} from '@gms/common-model/lib/filter/types';
import type { QcSegment } from '@gms/common-model/lib/qc-segment';
import type { SignalDetectionHypothesis } from '@gms/common-model/lib/signal-detection';
import type { Station } from '@gms/common-model/lib/station-definitions/station-definitions/station-definitions';

export interface UiChannelSegment {
  channelSegmentDescriptor: ChannelSegmentTypes.ChannelSegmentDescriptor;
  processingMasks: ChannelSegmentTypes.ProcessingMask[];
  channelSegment: ChannelSegmentTypes.ChannelSegment<ChannelSegmentTypes.TimeSeries>;
  domainTimeRange?: CommonTypes.TimeRange;
}

export type ChannelName = UiChannelSegment['channelSegmentDescriptor']['channel']['name'];
export type StationName = Station['name'];
/** Weavess row could be a channel name or station name */
export type WeavessRowName = ChannelName | StationName;
export type FilterName = FilterDefinition['name'];
/**
 * A combination of the namedFilter name (if it exists) + filter definition name
 */
export type FilterNameId = string;
export type SampleRate = FilterDefinition['filterDescription']['parameters']['sampleRateHz'];
/**
 * JSON.stringify({@link ChannelSegmentTypes.ChannelSegmentDescriptor})
 */
export type ChannelSegmentDescriptorId = string;
export type SignalDetectionIdString = SignalDetectionTypes.SignalDetection['id'];
export type SignalDetectionHypothesisId = SignalDetectionHypothesis['id']['id'];
export type EventId = Event['id'];
export type QcSegmentId = QcSegment['id'];

/**
 * Record<{@link ChannelName}, Record<{@link FilterName}, {@link UiChannelSegment}[]>>
 *
 * A record of channel names, each containing a record of filter names to an array of ui channel segments.
 */
export type UIChannelSegmentRecord = Record<ChannelName, Record<FilterName, UiChannelSegment[]>>;

/**
 * Record<JSON.stringify({@link ChannelSegmentTypes.ChannelSegmentDescriptor}), {@link ChannelSegment<FkTypes.FkPowerSpectra>}>>
 *
 * A record of Fk Channel Segments
 */
export type FkChannelSegmentRecord = Record<
  ChannelSegmentDescriptorId,
  ChannelSegmentTypes.ChannelSegment<FkTypes.FkPowerSpectra>
>;

/**
 * Record<JSON.stringify({@link SignalDetectionIdString}), {@link FkTypes.FkFrequencyThumbnail[]}>>
 *
 * A record of FkFrequencyThumbnails associated to a SignalDetection
 */
export type FkFrequencyThumbnailRecord = Record<
  SignalDetectionIdString,
  FkTypes.FkFrequencyThumbnail[]
>;

/**
 * Record<{@link ChannelName}, {@link ChannelTypes.Channel}>
 *
 * A record of channel names to a matching channel.
 */
export type ChannelRecord = Record<ChannelName, ChannelTypes.Channel>;

/**
 * Record<{@link WeavessRowName}, {@link FilterTypes.Filter}>
 *
 * A record of weavess row names (either a channel name or station name) to a filter.
 */
export type ChannelFilterRecord = Record<WeavessRowName, FilterTypes.Filter>;

/**
 * Record<{@link FilterName}, Record<{@link SampleRate}, {@link FilterDefinition}[]>>
 *
 * A record of filter names, each containing a record of sample rates to a filter definition.
 */
export type FilterDefinitionsRecord = Record<FilterName, Record<SampleRate, FilterDefinition>>;

/**
 * Record<{@link SignalDetectionHypothesisId}, {@link FilterDefinitionByFilterDefinitionUsage}>
 */
export type FilterDefinitionsForSignalDetectionsRecord = Record<
  SignalDetectionHypothesisId,
  FilterDefinitionByFilterDefinitionUsage
>;

/**
 * Record<JSON.stringify({@link ChannelSegmentTypes.ChannelSegmentDescriptor}), {@link FilterDefinitionByFilterDefinitionUsage}>
 */
export type FilterDefinitionsForChannelSegmentsRecord = Record<
  ChannelSegmentDescriptorId,
  FilterDefinitionByFilterDefinitionUsage
>;

/**
 * Record<{@link FilterNameId}, Record<{@link ChannelName}, Set<{@link ChannelSegmentDescriptorId}>>>
 *
 * A record of filter name ids (a combination of the namedFilter name + filter definition name), each
 * containing a record of channel names to a set of unique channel segment descriptor ids
 * (stringified channelSegmentDescriptors).
 */
export type ProcessedItemsCacheRecord = Record<
  FilterNameId,
  Record<ChannelName, Set<ChannelSegmentDescriptorId>>
>;

/**
 * Record<{@link SignalDetectionIdString}, {@link SignalDetectionTypes.SignalDetection}>
 *
 * A record of signal detection ids to signal detections.
 */
export type SignalDetectionsRecord = Record<
  SignalDetectionIdString,
  SignalDetectionTypes.SignalDetection
>;

/**
 * Record<{@link EventId}, {@link EventTypes.Event}>
 *
 * A record of event names to an event.
 */
export type EventsRecord = Record<EventId, EventTypes.Event>;

/**
 * Record<{@link SignalDetectionIdString}, {@link EventTypes.AssociationConflict}>
 *
 * A record of Association Conflicts.
 */
export type AssociationConflictRecord = Record<SignalDetectionIdString, AssociationConflict>;

/**
 * Record<{@link ChannelName}, Record<{@link QcSegmentId}, {@link QcSegment}>>
 *
 * A record of channel names containing a record of qc segment id's to qc segments.
 */
export type QcSegmentRecord = Record<ChannelName, Record<QcSegmentId, QcSegment>>;

/**
 * Used by {@link FilterAssociation} in conjunction
 * with {@link FilterDefinitionAssociationsObject}.
 * Represents the id/startTime of a ChannelSegment that was filtered using a
 * particular {@link FilterTypes.FilterDefinition}
 */
export interface WaveformIdentifier {
  channelSegmentId: string;
  startTime: number;
}

/**
 * Used in conjunction with {@link FilterDefinitionAssociationsObject}.
 * Represents a collection of ChannelSegments that were filtered with a
 * particular {@link FilterTypes.FilterDefinition}
 */
export interface FilterAssociation {
  definition: FilterTypes.FilterDefinition;
  waveformIdentifiers: WaveformIdentifier[];
}

/**
 * This object associates {@link WeavessTypes.DataSegment}s to
 * designed {@link FilterTypes.FilterDefinition} objects for the purpose of file exporting.
 */
export interface FilterDefinitionAssociationsObject {
  filterAssociations: FilterAssociation[];
  channelSegments: UiChannelSegment[];
}

/**
 * Record<{@link ChannelSegmentDescriptorId}, {@link SignalDetectionHypothesisId}>
 *
 * A record of channel segment descriptor strings to signal detection hypothesis id
 */
export type ChannelSegmentsToSignalDetectionHypothesisRecord = Record<
  ChannelSegmentDescriptorId,
  SignalDetectionHypothesisId
>;
