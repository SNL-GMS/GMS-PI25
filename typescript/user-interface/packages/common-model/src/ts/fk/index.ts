export * from './types';
export * as Util from './util';
