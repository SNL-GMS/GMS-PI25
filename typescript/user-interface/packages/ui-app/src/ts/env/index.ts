const ENV: Record<string, string> = Object.freeze((window as any).ENV);

/**
 * Used to determine if an environment variable is not set
 *
 * @param envVar to be tested
 * @returns true when no value is set
 */
export function isNotSet(envVar: string) {
  return (
    envVar === 'undefined' ||
    envVar === undefined ||
    envVar === null ||
    envVar === 'null' ||
    envVar.length === 0
  );
}

const GMS_KEYCLOAK = Object.freeze({
  GMS_KEYCLOAK_REALM: ENV?.GMS_KEYCLOAK_REALM,
  GMS_KEYCLOAK_URL: ENV?.GMS_KEYCLOAK_URL,
  GMS_KEYCLOAK_CLIENT_ID: ENV?.GMS_KEYCLOAK_CLIENT_ID,
  GMS_DISABLE_KEYCLOAK_AUTH: ENV?.GMS_DISABLE_KEYCLOAK_AUTH === 'true'
});

export const {
  GMS_KEYCLOAK_REALM,
  GMS_KEYCLOAK_URL,
  GMS_KEYCLOAK_CLIENT_ID,
  GMS_DISABLE_KEYCLOAK_AUTH
} = GMS_KEYCLOAK;
