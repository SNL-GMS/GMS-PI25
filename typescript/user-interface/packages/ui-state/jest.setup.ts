/* eslint-disable max-classes-per-file */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable import/first */
/* eslint-disable import/no-extraneous-dependencies */
import 'fake-indexeddb/auto';

import type { FilterDefinition } from '@gms/common-model/lib/filter/types';
import crypto from 'crypto';
import * as util from 'util';

// This is to fix an issue with test leaks related to createStateSyncMiddleware() in jsdom
process.env.GMS_DISABLE_REDUX_STATE_SYNC = 'true';

Object.defineProperty(global, 'crypto', {
  value: {
    subtle: {
      digest: async (algorithm: AlgorithmIdentifier, msg: Uint8Array) =>
        (crypto.webcrypto as any).subtle.digest(algorithm, msg) as Promise<any>
    }
  }
});

Object.defineProperty(window, 'TextEncoder', {
  writable: true,
  value: util.TextEncoder
});
Object.defineProperty(window, 'TextDecoder', {
  writable: true,
  value: util.TextDecoder
});
Object.defineProperty(global, 'TextEncoder', {
  writable: true,
  value: util.TextEncoder
});
Object.defineProperty(global, 'TextDecoder', {
  writable: true,
  value: util.TextDecoder
});

import Adapter from '@cfaester/enzyme-adapter-react-18';
import Enzyme, { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import type { GlobalWithFetchMock } from 'jest-fetch-mock';

const customGlobal: GlobalWithFetchMock = (global as unknown) as GlobalWithFetchMock;
// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
customGlobal.fetch = require('jest-fetch-mock');

customGlobal.fetchMock = customGlobal.fetch;

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
require('jest-canvas-mock');

Enzyme.configure({ adapter: new Adapter() });

const globalAny: any = global;

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
globalAny.fetch = require('jest-fetch-mock');

// Make Enzyme functions available in all test files without importing
globalAny.shallow = shallow;
globalAny.render = render;
globalAny.mount = mount;
globalAny.toJson = toJson;
globalAny.TextEncoder = util.TextEncoder;

globalAny.window = window;

class Worker {
  public url: string;

  public onmessage: (message: Record<string, unknown>, transfer?: Transferable[]) => void;

  public constructor(stringUrl: string) {
    this.url = stringUrl;
    this.onmessage = () => {
      // do nothing
    };
  }

  public postMessage(message: Record<string, unknown>, transfer?: Transferable[]) {
    this.onmessage(message, transfer);
  }
}
globalAny.window.Worker = Worker;

class MessagePort {
  public url: string;

  public onmessage: (message: Record<string, unknown>, transfer?: Transferable[]) => void;

  public postMessage(message: Record<string, unknown>, transfer?: Transferable[]) {
    this.onmessage(message, transfer);
  }
}

class SharedWorker {
  public url: string;

  public port = new MessagePort();

  public constructor(stringUrl: string) {
    this.url = stringUrl;

    this.port.onmessage = () => {
      // do nothing
    };
  }
}
globalAny.window.SharedWorker = SharedWorker;

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports,  import/no-extraneous-dependencies
require('jest-fetch-mock').enableMocks();

jest.mock('@gms/ui-wasm', () => {
  const isDesigned = jest.requireActual('@gms/ui-wasm/lib/gms-filters-provider/is-designed');

  const filter = async (
    filterDefinition: FilterDefinition,
    data: Float64Array
  ): Promise<Float64Array> => Promise.resolve(data);

  const design = async (filterDefinition: FilterDefinition): Promise<FilterDefinition> =>
    Promise.resolve(filterDefinition);

  const getBoundsForPositionBuffer = async (): Promise<{
    min: number;
    minSecs: number;
    max: number;
    maxSecs: number;
  }> => Promise.resolve({ min: 0, minSecs: 0, max: 10, maxSecs: 10 });

  return {
    design,
    filter,
    getBoundsForPositionBuffer,
    isDesigned: isDesigned.isDesigned
  };
});
