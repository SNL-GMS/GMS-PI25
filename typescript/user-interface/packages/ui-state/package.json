{
  "name": "@gms/ui-state",
  "version": "0.1.0",
  "description": "GMS UI State",
  "license": "UNLICENSED",
  "author": "GMS Development Team",
  "sideEffects": false,
  "module": "./lib/ui-state.js",
  "typings": "./lib/ui-state.d.ts",
  "files": [
    "lib"
  ],
  "scripts": {
    "clean:build": "yarn rimraf lib/ dist/ docs/ coverage/ bundle-analyzer/ artifacts/ .cache/ junit.xml",
    "clean": "npm-run-all -s clean:build",
    "build:eslint": "yarn eslint -c .eslintrc-config.yaml 'src/**/*.{ts,tsx}' --quiet --cache --cache-strategy content --cache-file .cache/.eslintcache",
    "build:eslint:test": "yarn eslint -c .eslintrc-config-test.yaml '{__tests__,__mocks__,__integration-tests__}/**/*.{ts,tsx}' --quiet --cache --cache-strategy content --cache-file .cache/.eslintcache-test",
    "build": "yarn tsc -b ./tsconfig-build.json",
    "build:test": "yarn tsc -b ./tsconfig-test.json",
    "watch": "yarn tsc -b ./tsconfig-build.json -watch",
    "docs": "yarn clean:docs && typedoc --out docs ./src/ --name ui-state --mode file --readme README.md --target ES6 --includeDeclarations --excludeExternals --excludeNotExported --hideGenerator",
    "sonar": "echo \"run 'yarn sonar' from the top-level directory\"",
    "test": "yarn test:dev",
    "test:dev:jest": "jest --maxWorkers=50% --verbose --bail",
    "test:prod:jest": "jest --maxWorkers=4 --silent --bail --coverageReporters=lcovonly --reporters=default --reporters=jest-junit",
    "test:dev": "npm-run-all -p build:test test:dev:jest",
    "test:prod": "npm-run-all -p build:test test:prod:jest",
    "version": "echo $npm_package_version"
  },
  "dependencies": {
    "@blueprintjs/core": "4.19.2",
    "@gms/common-model": "0.1.0",
    "@gms/common-util": "0.1.0",
    "@gms/ui-util": "0.1.0",
    "@gms/ui-wasm": "0.1.0",
    "@gms/ui-workers": "0.1.0",
    "@gms/weavess-core": "0.1.0",
    "@reduxjs/toolkit": "1.9.1",
    "axios": "0.22.0",
    "d3": "5.5.0",
    "dexie": "3.2.2",
    "immer": "9.0.6",
    "immutable": "4.0.0-rc.12",
    "lodash": "4.17.21",
    "moment": "2.26.0",
    "msgpack-lite": "0.1.26",
    "p-queue": "6.6.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-redux": "8.0.5",
    "redux": "4.1.2",
    "redux-devtools-extension": "2.13.9",
    "redux-logger": "3.0.6",
    "redux-state-sync": "3.1.4",
    "redux-thunk": "2.4.1",
    "workbox-core": "6.5.4",
    "workbox-precaching": "6.5.4",
    "workbox-routing": "6.5.4",
    "workbox-strategies": "6.5.4",
    "worker-rpc": "0.2.0"
  },
  "devDependencies": {
    "@blueprintjs/eslint-plugin": "2.2.0",
    "@cfaester/enzyme-adapter-react-18": "0.6.0",
    "@testing-library/jest-dom": "5.16.5",
    "@testing-library/react": "13.4.0",
    "@testing-library/react-hooks": "7.0.2",
    "@types/faker": "4.1.8",
    "@types/jest": "29.1.1",
    "@types/lodash": "4.14.149",
    "@types/node": "18.11.9",
    "@types/react": "18.0.26",
    "@types/react-dom": "18.0.9",
    "@types/redux-logger": "3.0.9",
    "@types/redux-mock-store": "1.0.3",
    "@types/redux-state-sync": "3.1.3",
    "@typescript-eslint/eslint-plugin": "5.33.0",
    "@typescript-eslint/parser": "5.33.0",
    "enzyme": "3.11.0",
    "enzyme-to-json": "3.4.3",
    "eslint": "8.22.0",
    "eslint-config-airbnb": "19.0.4",
    "eslint-config-prettier": "8.5.0",
    "eslint-import-resolver-typescript": "3.4.1",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jest": "27.0.4",
    "eslint-plugin-jsdoc": "39.3.6",
    "eslint-plugin-jsx-a11y": "6.6.1",
    "eslint-plugin-no-null": "1.0.2",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "eslint-plugin-prettier": "4.2.1",
    "eslint-plugin-react": "7.30.1",
    "eslint-plugin-react-hooks": "4.6.0",
    "eslint-plugin-simple-import-sort": "7.0.0",
    "eslint-plugin-unicorn": "43.0.2",
    "eslint-plugin-workspaces": "0.7.0",
    "fake-indexeddb": "4.0.0",
    "faker": "4.1.0",
    "immer": "9.0.6",
    "jest": "29.1.2",
    "jest-canvas-mock": "2.4.0",
    "jest-cli": "29.1.2",
    "jest-environment-jsdom": "29.1.2",
    "jest-fetch-mock": "3.0.3",
    "jest-junit": "12.2.0",
    "jsdom": "20.0.1",
    "jsdom-global": "3.0.2",
    "node-fetch": "2.6.0",
    "npm-run-all": "4.1.5",
    "prettier": "2.1.2",
    "react-test-renderer": "18.2.0",
    "redux-mock-store": "1.5.4",
    "rimraf": "3.0.0",
    "ts-jest": "29.1.1",
    "ts-jest-mock-import-meta": "0.12.0",
    "ts-node": "10.9.1",
    "typedoc": "0.15.3",
    "typescript": "5.1.6"
  },
  "jest-junit": {
    "suiteNameTemplate": "ui-state/{filename}",
    "classNameTemplate": "ui-state/{filename}/{classname}",
    "titleTemplate": "{title}"
  }
}
