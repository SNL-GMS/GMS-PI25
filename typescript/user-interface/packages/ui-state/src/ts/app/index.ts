export * from './api';
export * from './history';
export * from './hooks';
export * from './query';
export * from './redux-provider';
export * from './sanitization';
export * from './state';
export * from './store';
export * from './subscription';
export * from './util';
