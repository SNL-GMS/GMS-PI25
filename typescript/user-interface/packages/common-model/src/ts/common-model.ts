import * as BeamformingTemplateTypes from './beamforming-templates/types';
import * as ChannelSegmentTypes from './channel-segment/types';
import * as ColorTypes from './color/types';
import * as CommonTypes from './common/types';
import * as DataAcquisitionTypes from './data-acquisition/types';
import * as Displays from './displays/types';
import * as Endpoints from './endpoints/types';
import * as EventTypes from './event';
// required for legacy components
import * as LegacyEventTypes from './event/legacy';
import * as FacetedTypes from './faceted';
import * as FilterUtil from './filter/filter-util';
import * as FilterTypes from './filter/types';
import * as FkTypes from './fk';
import * as ProcessingMaskDefinitionTypes from './processing-mask-definitions/types';
// required for legacy components
import * as QcMaskTypes from './qc-mask';
import * as QcSegmentTypes from './qc-segment';
import * as SignalDetectionTypes from './signal-detection';
import * as SohTypes from './soh/types';
import * as ChannelTypes from './station-definitions/channel-definitions/channel-definitions';
import * as ResponseTypes from './station-definitions/channel-definitions/response-definitions';
import * as StationTypes from './station-definitions/station-definitions/station-definitions';
import * as ProcessingStationTypes from './station-processing/types';
import * as ReferenceStationTypes from './station-reference/types';
import * as SystemMessageTypes from './system-message/types';
import * as TimeTypes from './time';
import * as TypeUtil from './type-util/type-util';
import * as ConfigurationTypes from './ui-configuration/types';
import * as UserProfileTypes from './user-profile/types';
import * as WaveformTypes from './waveform/types';
import * as WorkflowTypes from './workflow/types';

export {
  BeamformingTemplateTypes,
  ChannelSegmentTypes,
  ChannelTypes,
  ColorTypes,
  CommonTypes,
  ConfigurationTypes,
  DataAcquisitionTypes,
  Displays,
  Endpoints,
  EventTypes,
  FacetedTypes,
  FilterTypes,
  FilterUtil,
  FkTypes,
  LegacyEventTypes,
  ProcessingMaskDefinitionTypes,
  ProcessingStationTypes,
  QcMaskTypes,
  QcSegmentTypes,
  ReferenceStationTypes,
  ResponseTypes,
  SignalDetectionTypes,
  SohTypes,
  StationTypes,
  SystemMessageTypes,
  TimeTypes,
  TypeUtil,
  UserProfileTypes,
  WaveformTypes,
  WorkflowTypes
};
