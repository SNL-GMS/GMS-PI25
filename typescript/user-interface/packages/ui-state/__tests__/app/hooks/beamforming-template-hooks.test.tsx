import {
  defaultStations,
  processingAnalystConfigurationData
} from '@gms/common-model/__tests__/__data__';
import { beamformingTemplatesByBeamTypeByStationByPhase } from '@gms/common-model/__tests__/__data__/beamforming-templates/beamforming-templates-data';
import { renderHook } from '@testing-library/react-hooks';
import { produce } from 'immer';
import React from 'react';
import { Provider } from 'react-redux';

import {
  useBeamformingTemplatesForEvent,
  useBeamformingTemplatesForFK,
  useBeamformingTemplatesForVisibleStationsAndFavoritePhases
} from '../../../src/ts/app/hooks/beamforming-template-hooks';
import type { AppState } from '../../../src/ts/app/store';
import { getStore } from '../../../src/ts/ui-state';
import { appState } from '../../test-util';

const providerStore = getStore();

function TestReduxWrapper({ children }) {
  return <Provider store={providerStore}>{children}</Provider>;
}

jest.mock('../../../src/ts/app/hooks/react-redux-hooks', () => {
  const actualRedux = jest.requireActual('../../../src/ts/app/hooks/react-redux-hooks');
  const mockUseAppDispatch = jest
    .fn()
    .mockImplementation(() => jest.fn(async () => Promise.resolve()));
  return {
    ...actualRedux,
    useAppDispatch: mockUseAppDispatch,
    useAppSelector: jest.fn((stateFunc: (state: AppState) => any) => {
      const state: AppState = produce(appState, draft => {
        draft.app.analyst.phaseSelectorFavorites = {
          'Seismic & Hydroacoustic': [
            'P',
            'Pn',
            'Pg',
            'pP',
            'S',
            'PKP',
            'PKPdf',
            'PKPbc',
            'PKPab',
            'PcP',
            'ScP',
            'Sn',
            'Lg',
            'Rg',
            'sP'
          ]
        };
        draft.data.beamformingTemplates = beamformingTemplatesByBeamTypeByStationByPhase;
      });
      return stateFunc(state);
    })
  };
});

jest.mock(
  '../../../src/ts/app/api/processing-configuration/processing-configuration-api-slice',
  () => {
    const actual = jest.requireActual(
      '../../../src/ts/app/api/processing-configuration/processing-configuration-api-slice'
    );
    return {
      ...actual,
      useGetProcessingAnalystConfigurationQuery: jest.fn(() => ({
        data: processingAnalystConfigurationData
      }))
    };
  }
);

jest.mock('../../../src/ts/app/hooks/station-definition-hooks', () => {
  const actual = jest.requireActual('../../../src/ts/app/hooks/station-definition-hooks');
  return {
    ...actual,
    useVisibleStations: jest.fn(() => defaultStations)
  };
});

describe('test hooks', () => {
  it('is defined', () => {
    expect(useBeamformingTemplatesForVisibleStationsAndFavoritePhases).toBeDefined();
    expect(useBeamformingTemplatesForFK).toBeDefined();
  });

  it('gets an FK template', () => {
    const { result } = renderHook(useBeamformingTemplatesForFK, { wrapper: TestReduxWrapper });
    expect(result.current.data).toMatchSnapshot();
  });

  it('gets an EVENT template', () => {
    const { result } = renderHook(useBeamformingTemplatesForEvent, { wrapper: TestReduxWrapper });
    expect(result.current.data).toMatchSnapshot();
  });
});
