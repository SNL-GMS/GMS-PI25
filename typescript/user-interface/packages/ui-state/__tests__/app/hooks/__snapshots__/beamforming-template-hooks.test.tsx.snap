// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test hooks gets an EVENT template 1`] = `
{
  "PDAR": {
    "P": {
      "beamDescription": {
        "beamSummation": "COHERENT",
        "beamType": "EVENT",
        "phase": "P",
        "samplingType": "SNAPPED",
        "twoDimensional": true,
      },
      "beamDuration": 300,
      "inputChannels": [
        {
          "effectiveAt": 1636503404,
          "name": "PDAR.PD01.SHZ",
        },
        {
          "effectiveAt": 1636503404,
          "name": "PDAR.PD02.SHZ",
        },
        {
          "effectiveAt": 1636503404,
          "name": "PDAR.PD03.SHZ",
        },
      ],
      "leadDuration": 5,
      "minWaveformsToBeam": 2,
      "orientationAngleToleranceDeg": 5,
      "sampleRateToleranceHz": 0.5,
      "station": {
        "effectiveAt": 1689026400,
        "name": "PDAR",
      },
    },
  },
}
`;

exports[`test hooks gets an FK template 1`] = `
{
  "PDAR": {
    "P": {
      "beamDescription": {
        "beamSummation": "COHERENT",
        "beamType": "EVENT",
        "phase": "P",
        "samplingType": "SNAPPED",
        "twoDimensional": true,
      },
      "beamDuration": 300,
      "inputChannels": [
        {
          "effectiveAt": 1636503404,
          "name": "PDAR.PD01.SHZ",
        },
        {
          "effectiveAt": 1636503404,
          "name": "PDAR.PD02.SHZ",
        },
        {
          "effectiveAt": 1636503404,
          "name": "PDAR.PD03.SHZ",
        },
      ],
      "leadDuration": 5,
      "minWaveformsToBeam": 2,
      "orientationAngleToleranceDeg": 5,
      "sampleRateToleranceHz": 0.5,
      "station": {
        "effectiveAt": 1689026400,
        "name": "PDAR",
      },
    },
  },
}
`;
